### ISSUE TITLE
css.lint.hexColorLength seems to be now irrelevant

### ISSUE THREAD

**Issue by andrecasal** [microsoft/vscode#81808]
Issue still occurs in Insiders v1.39.0

- VSCode Version: 1.39.0
- OS Version: macOS 10.14.6

Steps to Reproduce:

1. Set css.lint.hexColorLength to error or warning
2. Create a css file with a wrongly formatted color value
```
div {
    color: #ccc0003;
}
```

Expected behavior:
An error or warning (as per the `css.lint.hexColorLength` setting) coming from the hexColorLength setting.

Actual behavior:
A `css-propertyvaluexpected` error.

The `css.lint.hexColorLength` setting seems to no longer be relevant.

Does this issue occur when all extensions are disabled? Yes

**Comment by octref** [comment]
@aeschli What do you think would be a good fix? Do we parse invalid colors as `HexColor` so this linting rule could work?

**Comment by aeschli** [comment]
Related entry on stackoverflow: https://stackoverflow.com/questions/57640557/when-is-vs-codes-setting-css-lint-hexcolorlength-fired

**Comment by aeschli** [comment]
Yes, best is probably to create a HexColor for all values matching `#[1-9A-Fa-f]+`
We'd then need to make sure all consumers of HexColor can handle invalid length.

**Comment by andrecasal** [comment]
Would also be good to update the setting's description.
